### PR TITLE
drivers : edac: intel socfpga edac: added agilex5 EDAC driver

### PR DESCRIPTION
**Pull request by Smale-12048867**
Added EDAC driver support Hard Processor Subsystem (HPS) in Agilex5 Interl socfpga decive.

- Detects and manages SIngle Bit and Double Bit Errors detected in DDR attached to the device and RAM located in HPS peripherals.
- Performs Preventive maintenance (HPS reset) upon a DBE detection. 
- Supports shell commands to inject and test the SBE and DBE.

### CODE REVIEW

**Comment by finikorg** [comment]
@Smale-12048867 Generic comments: there is EDAC interface: https://docs.zephyrproject.org/latest/hardware/peripherals/edac/index.html 
Could you use that interface? Please do not have another `edac.h` headers, one is, hopefully, enough.

If you use that interface you do not need to re-write edac shell, etc.

There is also documentation about IBECC here: https://docs.zephyrproject.org/latest/hardware/peripherals/edac/ibecc.html . Please write similar for your implementation.

File: samples/subsys/shell/shell_module/boards/intel_socfpga_agilex5_socdk.conf
Comment (kartben) [inline samples/subsys/shell/shell_module/boards/intel_socfpga_agilex5_socdk.conf:1]: as mentioned in other recent PRs, please no conf/overlay in there that are not related to the shell module API sample. Thanks!

**Review by kartben — CHANGES_REQUESTED** [review]
